Sources/Boundaries/Store.swift

```diff
@@ -52,4 +39,85 @@ public final class Store<S, A> {
   public func subscribe<T>(state keyPath: KeyPath<S, T>, _ subscriber: @escaping (T) -> Void) {
     self.subscribe(subscriber <<< view(getting(keyPath)))
   }
+
+  // Runs the effects and dispatches the resulting actions.
+  private func interpret(_ effect: Cmd<E>) {
+    switch effect {
+    case let .execute(e):
+      if let action = self.execute(e) {
+        self.dispatch(action)
+      }
+
+    case .parallel:
+      DispatchQueue.global().async {
+        catOptionals(self.interpretedActions(effect))
```

Comment:
in parallel we go to a background queue, run all the effects in parallel and collect their resulting (optional) actions, remove the `nil`s, and dispatch the actions back on the main thread. 

the `interpretedActions` function does the work of interpreting effects without actually running the actions, just collects em into an array.

Reply:
Want to think this through a bit more. Don't think we want to push everything through `DispatchQueue.main`. Seems like the effects themselves should know which queue to execute on.

Reply:
yeah I think I made a comment somewhere that the `parallel` command should take a queue to operate on, but after the effect is done I believe the resulting action should be dispatched back to `main`...